ros1_foxglove_bridge/src/ros1_foxglove_bridge_nodelet.cpp

```diff
@@ -844,6 +859,29 @@ class FoxgloveBridge : public nodelet::Nodelet {
     }
   }
 
+  void fetchAsset(const std::string& uri, uint32_t requestId, ConnectionHandle clientHandle) {
+    foxglove::FetchAssetResponse response;
+    response.requestId = requestId;
+
+    try {
+      if (!isWhitelisted(uri, _assetUriWhitelistPatterns)) {
+        throw std::runtime_error("Asset URI not allowed: " + uri);
+      }
+
+      const resource_retriever::MemoryResource memoryResource = _resource_retriever.get(uri);
```

Comment:
Is there any way to limit the ability to access local files within a certain directory? It seems like a big security hole if this can access any file on the local machine.

Reply:
> Is there any way to limit the ability to access local files within a certain directory? It seems like a big security hole if this can access any file on the local machine.

Agreed, that's why I added the allowlist parameter. If you only want to specify certain package folders, you would have to list them manually.  `file://` URIs are by default not allowed.

Example
- `['package://robot_description/.+']` - allow only access to `robot_description` package folder
- `['package://.+[.]dae$','package://.+[.]stl$']` - only allow fetching of `dae` & `stl` files from package folders

Maybe the default `['package://.*']` is too permissive. Should I change it to only allow urdf, dae & stl files?

Reply:
Bummer: with the default `['package://.*']` one can specify relative paths to retrieve files that are not intended to be fetched as assets: 

`"uri": "package://ur_description/../../../secret.txt"`

Reply:
Yep `..` is what I was thinking of.
If there is some upstream infrastructure we depend on then maybe there's not a lot we can do about this, but we should at least provide people with a warning before they enable this feature.

Reply:
> If there is some upstream infrastructure we depend on then maybe there's not a lot we can do about this, but we should at least provide people with a warning in documentation before they enable this feature.

The default regex is now a lot stricter. Do you think that's sufficient or should we the asset feature by default?